docs/2.guide/5.best-practices/security.md

```diff
@@ -0,0 +1,306 @@
+    <p>Logged in since {{ session.loggedInAt }}</p>
+    <button @click="clear">Logout</button>
+  </div>
+  <div v-else>
+    <h1>Not logged in</h1>
+    <a href="/auth/github">Login with GitHub</a>
+    <!-- or open the OAuth route in a popup -->
+    <button @click="openInPopup('/auth/github')">Login with GitHub</button>
+  </div>
+</template>
+```
+
+Remember that this module only works with a Nuxt server running as it uses server API routes and it won't work with `nuxt generate`. You can anyway use Hybrid Rendering to pre-render pages of your application or disable server-side rendering completely.
+
+:read-more{title="Nuxt Auth Utils" to="https://github.com/atinux/nuxt-auth-utils"}
+
+### Nuxt Authorization
+
+[Nuxt Authorization](https://github.com/barbapapazes/nuxt-authorization) is a module offers a simple yet powerful method for managing authorization in a Nuxt application, applicable to both client and server. It is authentication system agnostic but can be seamlessly integrated with nuxt-auth-utils. It comes with the following features:
+
+* Works on both the client (Nuxt) and the server (Nitro)
+* Write abilities once and use them everywhere
+* Agnostic of the authentication layer
+* Use components to conditionally show part of the UI
+* Primitives are can be accessed for a full customization
+
+Abilities constitute the rules that embody the `authorization logic`. They take a user and one or more resources and yield a deny or an allow condition. While they can be grouped by resource, they remain independent of one another.
+
+```ts
+export const editBook = defineAbility((user: User, book: Book) => {
+  return user.id === book.authorId
+})
+```
+
+The `defineAbility` function acts as a factory that creates an ability but nothing beyond that. It needs to be used with bouncer functions. In a server endpoint, the `authorize` function can be employed to grant access to a resource based on the abilities
+
+```ts
+export default defineEventHandler(async (event) => {
+  await authorize(event, editBook)
+
+  const books = await db.query.books.findMany()
+
+  return books
+})
+```
+
+:read-more{title="Nuxt Authorization" to="https://soubiran.dev/posts/nuxt-going-full-stack-how-to-handle-authorization"}
+
+
+## Common problems
+
+When building more complex Nuxt applications, you will probably encounter some of the problems listed below. Understanding these problems and fixing them will help make your website more secure.
+
+### Returning entities with sensitive data
+
+**Problem**: When fetching the data from external source (i.e. Database) and returning it in the Nuxt API endpoint, you may return the whole table entity (such as User) to the browser and handle it there. This can lead to unintentional leaking of sensitive information that may not even be needed. 
+
+**Solution**: Utilize `pick` or use `interceptors` concepts for both `useAsyncData` and `useFetch` to sanitize what you get from the server without leaking confidential data.
```

Comment:
This isn't considered as a security improvement but more of a performance improvement. Because it allows to reduce the amount of payload in the SSR HTML
Sensitive data must always be filtered by the API and not by the consumer.

Reply:
Agree, better to have everything filtered in the API/backend side. But if it is not there (maybe a mistake on the API/Backend side) we can always pick just the information we need. That's what I had in mind when I was adding this point. But we can also remove it if you think that it won't bring value and cause confusion instead :)

Reply:
:+1: You can move it to the performance part. but it still need to rephrase/remove the next line.

> This can lead to unintentional leaking of sensitive information that may not even be needed. 

The API returning a whole table entity as a response is still a data leak. Even tho the Users may not see it, people with bad intentions will look at the response. An API is and should always  be considered as something public and accessible.